src/cell.cpp

```diff
@@ -585,6 +591,46 @@ Cell::contains_complex(Position r, Direction u, int32_t on_surface) const
   }
 }
 
+//==============================================================================
+// DAGMC Cell implementation
+//==============================================================================
+#ifdef DAGMC
+DAGCell::DAGCell() : Cell{} {};
+
+std::pair<double, int32_t> DAGCell::distance(Position p, Direction u, int32_t on_surface) const {
+  moab::ErrorCode rval;
+  moab::EntityHandle vol = dagmc_ptr_->entity_by_id(3, id_);
+  moab::EntityHandle hit_surf;
+  double dist;
+  rval = dagmc_ptr_->ray_fire(vol, p.xyz, u.xyz, hit_surf, dist);
```

Comment:
True. I did that originally and then decided on `union` because it would be more clear (in a way). 

Another option is to force alignment in that struct, but, as @smharper alluded to, I should look at performance first to make sure it matters.